packages/backend/roles/role.ts

```diff
@@ -0,0 +1,41 @@
+import Config from './config.json';
+
+export interface RoleConfig {
+  id: number;
+  name: string;
+  permissions: string[];
+}
+
+export default class RoleConfigCollection {
+  private static roles: RoleConfig[] = Config.roles;
+
+  static getRoleByName(name: string): RoleConfig {
+    const role = this.roles.find((r) => r.name === name);
+    if (!role) {
+      throw new Error(`Role ${name} does not exist`);
+    }
+    return role;
+  }
+
+  static getRoleByID(id: number): RoleConfig {
+    const role = this.roles.find((r) => r.id === id);
+    if (!role) {
+      throw new Error(`Role ${id} does not exist`);
+    }
+    return role;
+  }
+
+  static getRoles(): RoleConfig[] {
+    return this.roles;
+  }
+
+  static getPermissions(): string[] {
+    return this.roles.flatMap((role) => role.permissions);
```

Comment:
Would probably be good to dedup this... you're gonna end up with a list of duplicates in the list I think

```js
const flatMapPermissions = this.roles.flatMap((role) => role.permissions);
return [...new Set(flatMapPermissions)];
```

Reply:
`Set` is pretty nifty. ill do this

Reply:
oh wait i dropped this function. Ill keep that in mind in the future though.